src/datamodule.py

```diff
@@ -87,38 +87,51 @@ def __init__(
         self.batch_size: int = batch_size
         self.num_workers: int = num_workers
 
-    def setup(self, stage: str | None = None):
+    def setup(self, stage: Literal["fit", "predict"] | None = None):
         """
         Data operations to perform on every GPU.
         Split data into training and test sets, etc.
+
+        Parameters
+        ----------
+        stage : str or None
+            Whether to setup the datapipe for the training/validation loop, or
+            the prediction loop. Choose from either 'fit' or 'predict'.
         """
-        # Step 1 - Get list of GeoTIFF filepaths from data/ folder
-        dp_paths = torchdata.datapipes.iter.FileLister(
-            root=self.data_path, masks="*.tif", recursive=True, length=423
-        )
+        # Step 1 - Get list of GeoTIFF filepaths from s3 bucket or data/ folder
+        if self.data_path.startswith("s3://"):
+            dp = torchdata.datapipes.iter.IterableWrapper(iterable=[self.data_path])
+            self.dp_paths = dp.list_files_by_s3(masks="*.tif")
+        else:  # if self.data_path is a local data path
+            self.dp_paths = torchdata.datapipes.iter.FileLister(
+                root=self.data_path, masks="*.tif", recursive=True
+            )
```

Comment:
Note that loading data from s3 is still a bit slower than a local data folder, but it can be useful for cases like one-off inference/prediction when we don't want to store large amounts of data locally.